pkg/api/resolver_query_scraper.go

```diff
@@ -6,114 +6,178 @@ import (
 	"fmt"
 	"strconv"
 
-	"github.com/stashapp/stash/pkg/manager"
 	"github.com/stashapp/stash/pkg/manager/config"
 	"github.com/stashapp/stash/pkg/models"
 	"github.com/stashapp/stash/pkg/scraper"
 	"github.com/stashapp/stash/pkg/scraper/stashbox"
 )
 
-// deprecated
-func (r *queryResolver) ScrapeFreeones(ctx context.Context, performer_name string) (*models.ScrapedPerformer, error) {
-	scrapedPerformer := models.ScrapedPerformerInput{
-		Name: &performer_name,
-	}
-	return manager.GetInstance().ScraperCache.ScrapePerformer(scraper.FreeonesScraperID, scrapedPerformer)
+func (r *queryResolver) ScrapeURL(ctx context.Context, url string, ty models.ScrapeContentType) (models.ScrapedContent, error) {
+	return r.scraperCache.ScrapeURL(ctx, url, ty)
 }
 
 // deprecated
 func (r *queryResolver) ScrapeFreeonesPerformerList(ctx context.Context, query string) ([]string, error) {
-	scrapedPerformers, err := manager.GetInstance().ScraperCache.ScrapePerformerList(scraper.FreeonesScraperID, query)
+	content, err := r.scraperCache.ScrapeName(ctx, scraper.FreeonesScraperID, query, models.ScrapeContentTypePerformer)
+
+	if err != nil {
+		return nil, err
+	}
 
+	performers, err := marshalScrapedPerformers(content)
 	if err != nil {
 		return nil, err
 	}
 
 	var ret []string
-	for _, v := range scrapedPerformers {
-		if v.Name != nil {
-			ret = append(ret, *v.Name)
+	for _, p := range performers {
+		if p.Name != nil {
+			ret = append(ret, *p.Name)
 		}
 	}
 
 	return ret, nil
 }
 
+func (r *queryResolver) ListScrapers(ctx context.Context, types []models.ScrapeContentType) ([]*models.Scraper, error) {
+	// Collect all scrapers in a map so they occur at-most-once
+	m := make(map[string]*models.Scraper)
+	for _, ty := range types {
+		scrapers := r.scraperCache.ListScrapers(ty)
+		for _, s := range scrapers {
+			if _, ok := m[s.ID]; !ok {
+				m[s.ID] = s
+			}
+		}
+	}
+
+	var ret []*models.Scraper
+	for _, v := range m {
+		ret = append(ret, v)
+	}
+
```

Comment:
After scrapers are deduped from the map the order is lost. The current dev version prints the scrapers in the order they were loaded ( coincides with a by name sort in most cases). I think we could sort the `ret` here by the `Name` field before returning it. This makes the lists look better either in the settings page and in the scrape with actions.

Reply:
Good call, I agree.